Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -316,18 +321,28 @@ public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.Event
         /// Assumption: We only send pointer events to the objects that pointers are focusing, except for global event listeners (which listen to everything)
         /// In contract, all other events get sent to all other pointers attached to a given input source
         /// </summary>
-        private void HandlePointerEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<T> eventHandler) where T : IMixedRealityPointerHandler
+        private void HandlePointerEvent(
+            MixedRealityPointerEventData eventData, ExecuteEvents.EventFunction<IMixedRealityPointerHandler> eventHandler)
         {
             if (disabledRefCount > 0)
             {
                 return;
             }
 
             Debug.Assert(eventData != null);
-            var baseInputEventData = ExecuteEvents.ValidateEventData<BaseInputEventData>(eventData);
-            DispatchEventToGlobalListeners(baseInputEventData, eventHandler);
+            DispatchEventToGlobalListeners(eventData, eventHandler);
 
-            if (baseInputEventData.used)
+            if (pointerHandlers != null)
+            {
+                eventExecutionDepth++;
+                foreach (var handler in pointerHandlers)
```

Comment:
>foreach [](start = 16, length = 7)

I've avoided using foreach in inner loop stuff before because of past guidance around the speed of for vs foreach, and the fact that foreach generates GC garbage but for does not.

Granted I think that this advice came around from older Unity versions which created significant garbage for each run - I see that there are still some recent guidelines for plain old C# talking about preferring for over foreach, but again because we do IL2CPP it's not clear to me if that actually has any real effect now. So this is probably fine at this point, if we wanted to be paranoid we could do the list thing that BaseEventSystem does.

I don't really have a preference, so feel free to just just ACK this.

Reply:
I don't think `foreach` generates GC here. We can go over it if you want. I think the general guideline should be the opposite: use `foreach` unless you know for a fact the loop is hot and `for` will make a significant difference in performance.